### PR TITLE
New wallets compatible with bot interface

### PR DESCRIPTION
**Pull request by mikekeke**
Some steps to make integration with `bot-plutus-interface` possible:
- during creation of wallet `.skey` is generated and saved to known folder
- during cluster setup protocol parameters queried and saved to known folder

### CODE REVIEW

File: src/BotInterface/Wallet.hs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+module BotInterface.Wallet (
+  BpiWallet,
+  usingEnv,
+  addSomeWallet,
+  mkMainnetAddress,
+  cardanoMainnetAddress,
+) where
+
+import BotInterface.Setup qualified as Setup
+import BotInterface.Types
```
Comment (gege251) [inline src/BotInterface/Wallet.hs:10]: This should be a qualified import or explicit import list

File: src/Tools/CardanoApi.hs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import Cardano.Api qualified as C
+import Cardano.Api.ProtocolParameters (ProtocolParameters)
+import Cardano.Launcher.Node (nodeSocketFile)
+import Cardano.Slotting.Slot (WithOrigin)
+import Cardano.Wallet.Shelley.Launch.Cluster (RunningNode (..))
+import Control.Exception (Exception)
+import Data.Set qualified as Set
+import GHC.Generics (Generic)
+import LocalCluster.Types
```
Comment (gege251) [inline src/Tools/CardanoApi.hs:15]: And this one also

**Review by gege251 — COMMENTED** [review]
Added a few suggestions

File: example/Main.hs [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-import LocalCluster.CardanoApi qualified as LCAPI
 import LocalCluster.Cluster (runUsingCluster)
-import LocalCluster.DebugCli qualified as CLI
 import LocalCluster.Wallet
 import System.Environment (setEnv)
+import Tools.CardanoApi qualified as LCAPI
+import Tools.DebugCli qualified as CLI
 import Utils (ada, waitSeconds)
 
 import Address as Addr
```
[the remark this answers is not in the document]
Comment (mikekeke) [reply example/Main.hs]: That was unused import not picked by linter for some reason

File: example/Main.hs [hunk 4]
```diff
@@ -22,36 +25,19 @@ main = do
...
-      >>= print
-    debugCheck cEnv (singleWallet : wallets)
-
-
+    putStrLn "\nDebug check:"
+    putStrLn $ "Cluster dir: " <> show (supportDir cEnv)
+    waitSeconds 2
+    case ws of
+      Left e -> error $ "Err: " <> show e
+      Right ws' -> mapM_ (CLI.utxoAtAddress cEnv . BW.mkMainnetAddress) ws'
```
[the remark this answers is not in the document]
Comment (mikekeke) [reply example/Main.hs]: Suggestion applied

File: example/Main.hs [hunk 5]
```diff
@@ -1,17 +1,20 @@
...
 import LocalCluster.Cluster (runUsingCluster)
-import LocalCluster.DebugCli qualified as CLI
 import LocalCluster.Wallet
 import System.Environment (setEnv)
+import Tools.CardanoApi qualified as LCAPI
+import Tools.DebugCli qualified as CLI
 import Utils (ada, waitSeconds)
 
 import Address as Addr
+import BotInterface.Types
```
[the remark this answers is not in the document]
Comment (mikekeke) [reply example/Main.hs]: addressed

File: src/BotInterface/Wallet.hs (hunk 1 above)
Comment (mikekeke) [reply src/BotInterface/Wallet.hs]: addressed

File: src/Tools/CardanoApi.hs (hunk 2 above)
Comment (mikekeke) [reply src/Tools/CardanoApi.hs]: addressed

**Comment by mikekeke** [comment]
@gege251 Thanks!

**Pull request merged**